### PR TITLE
(PC-11697) feat(changeEmailError): add cta to restart the email change flow

### PR DESCRIPTION
**Pull request by voisinhugo**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-11697

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
| image |         | image | image |                  |
| image |  | image | image |  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/profile/pages/ChangeEmail/__tests__/ChangeEmailExpiredLink.native.test.tsx [hunk 1]
```diff
@@ -55,11 +69,21 @@ describe('<ChangeEmailExpiredLink />', () => {
...
+  })
 })
 
 function renderChangeEmailExpiredLink() {
-  const navigationProps = {
-    route: { params: { email: userEmail } },
-  } as StackScreenProps<RootStackParamList, 'ChangeEmailExpiredLink'>
-  return render(<ChangeEmailExpiredLink {...navigationProps} />)
+  return render(<ChangeEmailExpiredLink />)
 }
```
Comment (Lucasbeneston) [inline src/features/profile/pages/ChangeEmail/__tests__/ChangeEmailExpiredLink.native.test.tsx:89]: Tu n'as plus besoin de `renderChangeEmailExpiredLink`, perso j'aurais mis juste les render(<...>) direct dans les tests

File: src/features/profile/pages/ChangeEmail/__tests__/ChangeEmailExpiredLink.web.test.tsx [hunk 2]
```diff
@@ -27,22 +40,8 @@ describe('<ChangeEmailExpiredLink />', () => {
...
-  })
 })
 
 function renderChangeEmailExpiredLink() {
-  const navigationProps = {
-    route: { params: { email: userEmail } },
-  } as StackScreenProps<RootStackParamList, 'ChangeEmailExpiredLink'>
-  return render(<ChangeEmailExpiredLink {...navigationProps} />)
+  return render(<ChangeEmailExpiredLink />)
 }
```
Comment (Lucasbeneston) [inline src/features/profile/pages/ChangeEmail/__tests__/ChangeEmailExpiredLink.web.test.tsx:47]: idem

**Review by Lucasbeneston — APPROVED** [review]
Top j'ai juste un mini changement à proposer sur les tests 👍

File: src/features/auth/forgottenPassword/ResetPasswordExpiredLink/ResetPasswordExpiredLink.tsx [hunk 3]
```diff
@@ -35,10 +37,17 @@ export function ResetPasswordExpiredLink(props: Props) {
     }
   }
 
+  const renderResendEmailButton = () => (
+    <ButtonPrimaryWhite
+      title={t`Renvoyer l'email`}
+      onPress={() => resetPasswordEmailQuery()}
```
Comment (antoinewg) [inline src/features/auth/forgottenPassword/ResetPasswordExpiredLink/ResetPasswordExpiredLink.tsx:43]:
```suggestion
      onPress={resetPasswordEmailQuery}
```

le mieux est d'éviter de passer une nouvelle fonction quand c'est possible.

File: src/features/auth/signup/SignupConfirmationExpiredLink/SignupConfirmationExpiredLink.tsx [hunk 4]
```diff
@@ -35,10 +37,17 @@ export function SignupConfirmationExpiredLink(props: Props) {
     }
   }
 
+  const renderResendEmailButton = () => (
+    <ButtonPrimaryWhite
+      title={t`Renvoyer l'email`}
+      onPress={() => signupConfirmationExpiredLinkQuery()}
```
Comment (antoinewg) [inline src/features/auth/signup/SignupConfirmationExpiredLink/SignupConfirmationExpiredLink.tsx:43]: idem

File: src/features/profile/pages/ChangeEmail/ChangeEmailExpiredLink.tsx [hunk 5]
```diff
@@ -1,41 +1,42 @@
...
-    ? t`Tu peux te connecter sur ton profil pour recommencer le parcours de changement d’e-mail.` +
-      '\n' +
-      '\n' +
-      t`Si tu as besoin d’aide, n’hésite pas à contacter le support.`
-    : undefined
+  const bodyText =
+    t`Ton adresse e-mail n’a pas été modifiée. Le lien que tu reçois par e-mail expire 24h après sa réception.` +
+    '\n' +
+    '\n' +
+    (isLoggedIn
```
Comment (antoinewg) [inline src/features/profile/pages/ChangeEmail/ChangeEmailExpiredLink.tsx:27]: tu pense quoi de définir 2 variables: `bodyText`, `bodySubText` et `customBodyText={bodyText + "\n\n" + bodySubText}`

**Pull request merged**